src/os/bluestore/BlueFS.cc

```diff
@@ -4738,6 +4738,28 @@ size_t BlueFS::probe_alloc_avail(int dev, uint64_t alloc_size)
   }
   return total;
 }
+
+void BlueFS::trim_free_space(const string& type)
+{
+  unsigned bdev_id;
+  if(type == "bdev-wal") {
+    bdev_id = BDEV_WAL;
+  } else if (type == "bdev-db") {
+    bdev_id = BDEV_DB;
+  } else {
+    derr << __func__ << "unknown bdev type " << type << dendl;
+    return;
+  }
+  auto iterated_allocation = [&](size_t off, size_t len) {
+    ceph_assert(len > 0);
+    bdev[bdev_id]->discard(off, len);
+  };
+  if (alloc[bdev_id] &&
```

Comment:
We need separate messages for:
1) bdev[bdev_id] is not configured
2) bdev[bdev_id] does not support dicard.
Both to cout, as it is to printed to ceph-bluestore-tool output.

Reply:
done